src/sentry/static/sentry/app/views/performance/vitalDetail/index.tsx

```diff
@@ -0,0 +1,134 @@
+import {t} from 'app/locale';
+import {loadOrganizationTags} from 'app/actionCreators/tags';
+import {Organization, Project, GlobalSelection} from 'app/types';
+import SentryDocumentTitle from 'app/components/sentryDocumentTitle';
+import GlobalSelectionHeader from 'app/components/organizations/globalSelectionHeader';
+import {PageContent} from 'app/styles/organization';
+import EventView from 'app/utils/discover/eventView';
+import {WebVital} from 'app/utils/discover/fields';
+import {decodeScalar} from 'app/utils/queryString';
+import LightWeightNoProjectMessage from 'app/components/lightWeightNoProjectMessage';
+import withApi from 'app/utils/withApi';
+import withGlobalSelection from 'app/utils/withGlobalSelection';
+import withOrganization from 'app/utils/withOrganization';
+import withProjects from 'app/utils/withProjects';
+
+import {addRoutePerformanceContext, getTransactionName} from '../utils';
+import VitalDetailContent from './vitalDetailContent';
+import {generatePerformanceVitalDetailView} from '../data';
+
+type Props = {
+  api: Client;
+  location: Location;
+  params: Params;
+  organization: Organization;
+  projects: Project[];
+  selection: GlobalSelection;
+  loadingProjects: boolean;
+  router: InjectedRouter;
+};
+
+type State = {
+  eventView: EventView | undefined;
+};
+
+class VitalDetail extends React.Component<Props, State> {
+  state: State = {
+    eventView: generatePerformanceVitalDetailView(
+      this.props.organization,
+      this.props.location
+    ),
+  };
+
+  static getDerivedStateFromProps(nextProps: Props, prevState: State): State {
+    return {
+      ...prevState,
+      eventView: generatePerformanceVitalDetailView(
+        nextProps.organization,
+        nextProps.location
+      ),
+    };
+  }
+
+  componentDidMount() {
+    const {api, organization, selection} = this.props;
+    loadOrganizationTags(api, organization.slug, selection);
+    addRoutePerformanceContext(selection);
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const {api, organization, selection} = this.props;
+
+    if (
+      !isEqual(prevProps.selection.projects, selection.projects) ||
+      !isEqual(prevProps.selection.datetime, selection.datetime)
+    ) {
+      loadOrganizationTags(api, organization.slug, selection);
+      addRoutePerformanceContext(selection);
+    }
+  }
+
+  getDocumentTitle(): string {
+    const name = getTransactionName(this.props.location);
+
+    const hasTransactionName = typeof name === 'string' && String(name).trim().length > 0;
+
+    if (hasTransactionName) {
+      return [String(name).trim(), t('Performance')].join(' - ');
+    }
+
+    return [t('Vital Detail'), t('Performance')].join(' - ');
+  }
+
+  render() {
+    const {organization, location, router} = this.props;
+    const {eventView} = this.state;
+    if (!eventView) {
+      browserHistory.replace({
+        pathname: `/organizations/${organization.slug}/performance/`,
+        query: {
+          ...location.query,
```

Comment:
Should the location be modified? If the previous eventView was invalid, what will redirecting back to the same view with the same parameters fix?

Reply:
We use this pattern in a few places, it's both the null guard and a redirect to performance landing from the detail view though?